### ISSUE TITLE
Unable to disable source maps

### ISSUE THREAD

**Issue by stefee** [swc-project/cli#73]
Reproduction using `@swc/cli@0.1.51`.

`scripts/.swcrc` contents:

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript"
    },
    "target": "es2021"
  }
}

```

Steps:

1. `node_modules/.bin/swc promiseSettledAggregate.ts -o dist/promiseSettledAggregate.js --config-file scripts/.swcrc`
2. See added `promiseSettledAggregate.js.map` in `dist`
3. Delete `dist/promiseSettledAggregate.js.map`
4. Run `npm install @swc/cli@0.1.50` and re-run first step, and see now there is no longer `promiseSettledAggregate.js.map`

Expected behaviour:

Only emit `promiseSettledAggregate.js.map` if `sourceMaps` is enabled in config file. Do not emit if `sourceMaps` is omitted or set to `false`.

**Comment by stefee** [comment]
It seems the issue is swc is not reading `sourceMaps` boolean value correctly from `.swcrc` - only `"inline"` has effect, but not `true` or `false` - sourcemaps will always be generated unless `"inline"` is specified.

I have found that this issue is consistent whether using `--config-file` flag on CLI or not.

**Comment by stefee** [comment]
I haven't had time to investigate this yet but I am still seeing this bug on latest version @kdy1 - do you know the cause?

**Comment by kdy1** [comment]
No, I don't know how `sourceMaps` can be true while there's no `.swcrc` and `sourceMaps` is not specified while invoking `@swc/core`
